### DISCUSSION TITLE
There is a way to hide callbackUrl from url?

### DISCUSSION THREAD

**Discussion by flosrn** [nextauthjs/next-auth#3846] [Help]
I am using a custom signin page and I don't want the user to see the callbackUrl in signing page url (currently `http://localhost:3000/auth/signin?callbackUrl=http://localhost:3000/`)

It would be better for me if the url to be just `http://localhost:3000/auth/signin`.

And error only `http://localhost:3000/auth/error?error=Callback` 

Currently, the url for the error is `http://localhost:3000/auth/signin?callbackUrl=http://localhost:3000/auth/signin&error=Callback` so it was really weird with duplicate origin.

How to avoid that please and just have a clean url with clean error without duplicate origin?

**Comment by pedroimpulcetto-ciandt** [comment]
The same problem with callbackUrl after the custom sign-in page in another discussion.  
- #3748

**Comment by DonHuskini** [comment]
Yes, the best way I found to do it is to call the endpoint instead of `signIn()` method. It's the same thing.
`router.push('/api/auth/signin', { query: {} })`  // url is /login
`router.push('/api/auth/signin', { query: { state: 'new-user' } })` // url is /login?state=new-user

**Comment by ConnectWithNoor** [comment]
Is their any work around for the problem? I'm too stuck in the same issue

**Reply by korvus** [reply]
For error only, one way to fix it is to do this:

In pages/api/auth/ create a file with this name: error.js / .ts
Then, just make a redirect the way you want that way: 

```javascript
export default async function (req, res) {
    const { error } = req.query;
    const redirectUrl = `/?error=${error}`;
    res.redirect(redirectUrl);
}
```
That's all.
